src/util.c

```diff
@@ -321,6 +333,31 @@ char *git__strsep(char **end, const char *sep)
 	return NULL;
 }
 
+
+const char *git__strnchr(const char *str, size_t len, char c)
+{
+	for (; len; len--) {
+		if (str[len] == c)
+			return &str[len];
+	}
+
+	return NULL;
+}
```

Comment:
How is this different from `memchr()` other than the return type and order of arguments?